src/Database/Database.php

```diff
@@ -379,8 +386,14 @@ public function query(string $query, array $bindings = [], array $params = [])
 
 		// apply the fetch closure to all results if given
 		if ($options['fetch'] instanceof Closure) {
-			foreach ($results as $key => $result) {
-				$results[$key] = $options['fetch']($result, $key);
+			if ($options['method'] === 'fetchAll') {
+				// fetching multiple records
+				foreach ($results as $key => $result) {
+					$results[$key] = $options['fetch']($result, $key);
+				}
+			} elseif ($options['method'] === 'fetch' && $results !== false) {
+				// fetching a single record
+				$results = $options['fetch']($results);
```

Comment:
I think we need to explicitly pass `null` as the second argument, otherwise we will get "too few arguments" errors when using a closure or callable that expects two arguments.